(1-3)中級 Eメール重複チェック　機能の作成

HayatoSaishu opened the pull request:
メールアドレスの重複チェック機能を作成しました。

問題があればコメントを頂ければと思います。
特に問題がなければマージをお願い致します。

Commit:
メールアドレスの重複チェック機能の作成

igamasayuki reviewed (changes requested):
　今回のエラーメッセージの登録ですが、
　model.addAttribute("キー", "エラーメッセージ");
　ではなく、resultにエラーメッセージを追加することができます。
　このようにすると、viewでは
　「○○を入力してください」
　というformのvalidateエラーが出る箇所に、今回のものと共通でエラーメッセージを出すことができます。
　BindingResultのAPIマニュアルを見ながら探してみてください。

igamasayuki committed:
Merge branch 'develop' into feature/emailCheck

Commit:
BindingResultにメールアドレス重複時のエラーメッセージを追加

Commit:
変更点の修正
Merge branch 'feature/emailCheck' of https://github.com/HayatoSaishu/ex-emp-management-bugfix into feature/emailCheck

Commit:
@Emailの入力値チェックを追加

igamasayuki commented on src/main/java/jp/co/sample/emp_management/controller/AdministratorController.java:
```diff
@@ -72,15 +73,15 @@ public String toInsert() {
 			return "employee/list";
 		} else {
-			model.addAttribute("errorMessage", "メールアドレスが既に登録されています");
+			result.rejectValue("mailAddress", "mailAddress", "既にメールアドレスが登録されています");
```
この処理で良いのですが、

if(result.hasErrors){}がないので、フォームで設定した入力値エラーが出ないような気がしますがいかがでしょうか？

正しい場合：修正後再プッシュをお願いします。

私の認識が違う場合：Slackにて連絡をお願いします。

igamasayuki reviewed (changes requested):
気になる点をコメントしました。
差し戻しますので、再度pushして管理シートより再レビュー依頼をしてください。
(この場合、push後、再度プルリクを作る必要はありません。)

Commit:
developブランチとマージ
Merge branch 'develop' into feature/emailCheck

Commit:
管理者登録時に入力値エラーがあった場合、エラーメッセージが出るように修正

Commit:
入力値エラーとメールアドレスの重複エラーが同じタイミングで表示されるように修正

Commit:
入力値エラーが出た際にメールアドレスの重複エラーも出てしまう問題を修正

igamasayuki reviewed (changes requested):
とても苦労されたと思うので答えをお伝えしますが、

1-3と1-5は以下のような感じにするととてもスッキリします。
修正後再度pushをお願いします。

		// パスワード確認
		if(!form.getPassword().equals(form.getConfirmationPassword())){
			result.rejectValue("password", "", "パスワードが一致していません");
			result.rejectValue("confirmationPassword", "", "");
		}

		// メールアドレスが重複している場合の処理
		Administrator existAdministrator = administratorService.findByMailAddress(form.getMailAddress());
		if(existAdministrator != null){
			result.rejectValue("mailAddress", "", "そのメールアドレスは既に登録されています");
		}
		
		// エラーが一つでもあれば入力画面に戻る
		if (result.hasErrors()) {
			  return toInsert();
		}
